Problem with upgrade from 3.0.0.Alpha3 to 3.0.0 in project with Quarkus BOM

Recent effort to upgrade `quarkiverse.minio` client from `3.0.0.Alpha` to `3.0.0` in Camel Quarkus is not possible.

1. Problem happens in case that the maven module uses Quarkus BOM and not explicit dependency versions (which is the case of camel-quarkus)
2. Quarkus BOM (3.0.1.Final currently) forces versions of okhttp (3.14.9) and okio (1.17.2)
3. Quarkiverse.minio 3.0.0 bring minio 8.5.2 which requires okhttp >= 4.8.1 (see the [code](https://github.com/minio/minio-java/blob/master/api/src/main/java/io/minio/S3Base.java#L106))

All items combined makes upgrade from 3.0.0.Alpha3 to 3.0.0 impossible.

I created the small reproducer, where the error can be simulated (without camel-quarkus) https://github.com/JiriOndrusek/minioClientQuarkiverseReproducer

I think I'm seeing the same or similar problem. Here are Quarkus startup failure logs on `compile:dev` with a Minio devservice:

```
$ ./mvnw compile quarkus:dev

[output trimmed]

2023-04-28 17:22:27,329 INFO  [org.hib.too.sch.int.exe.GenerationTargetToDatabase] (JPA Startup Thread) HHH000476: Executing script '[injected ScriptSourceInputNonExistentImpl script]'
2023-04-28 17:22:27,408 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile [dev]): java.lang.NoSuchMethodError: 'okhttp3.RequestBody okhttp3.RequestBody.create(byte[],okhttp3.MediaType)'
	at io.minio.S3Base.<clinit>(S3Base.java:104)
	at io.minio.MinioClient$Builder.<init>(MinioClient.java:2447)
	at io.minio.MinioClient.builder(MinioClient.java:2439)
	at io.quarkiverse.minio.client.MinioClients.createMinioClient(MinioClients.java:70)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at io.quarkiverse.minio.client.MinioClients.getMinioClient(MinioClients.java:53)
	at io.quarkiverse.minio.client.MinioClients.fromName(MinioClients.java:49)
	at io.quarkiverse.minio.client.MinioRecorder.lambda$minioClientSupplier$0(MinioRecorder.java:13)
	at io.quarkus.arc.runtime.ArcRecorder$4.apply(ArcRecorder.java:129)
	at io.quarkus.arc.runtime.ArcRecorder$4.apply(ArcRecorder.java:126)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.createSynthetic(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.doCreate(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.create(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.get(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.get(Unknown Source)
	at io.cryostat.recordings.Recordings_Bean.doCreate(Unknown Source)
	at io.cryostat.recordings.Recordings_Bean.create(Unknown Source)
	at io.cryostat.recordings.Recordings_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
	at io.cryostat.recordings.Recordings_Observer_onStart_d7d5577ed7c7cf36fa3bfe9393292221b1355418.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:346)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:328)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:82)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:155)
	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:106)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:104)
	at java.base/java.lang.Thread.run(Thread.java:833)


2023-04-28 17:22:27,410 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2023-04-28 17:22:27,433 INFO  [io.qua.qui.dep.pac.PackageManager] (main) Stopping Quinoa package manager live coding as a dev service.
2023-04-28 17:22:27,433 INFO  [io.qua.qui.dep.pac.PackageManager] (main) Stopping Quinoa package manager live coding as a dev service.
2023-04-28 17:22:28,383 INFO  [io.qua.dev.com.ContainerShutdownCloseable] (main) Dev Services for PostgreSQL shut down.
2023-04-28 17:22:28,383 INFO  [io.qua.dev.com.ContainerShutdownCloseable] (main) Dev Services for PostgreSQL shut down.
2023-04-28 17:22:28,391 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: io.quarkus.dev.appstate.ApplicationStartException: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.dev.appstate.ApplicationStateNotification.waitForApplicationStart(ApplicationStateNotification.java:58)
	at io.quarkus.runner.bootstrap.StartupActionImpl.runMainClass(StartupActionImpl.java:123)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:137)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:447)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:59)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:149)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:104)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:104)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ExceptionInInitializerError
	at io.minio.MinioClient$Builder.<init>(MinioClient.java:2447)
	at io.minio.MinioClient.builder(MinioClient.java:2439)
	at io.quarkiverse.minio.client.MinioClients.createMinioClient(MinioClients.java:70)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at io.quarkiverse.minio.client.MinioClients.getMinioClient(MinioClients.java:53)
	at io.quarkiverse.minio.client.MinioClients.fromName(MinioClients.java:49)
	at io.quarkiverse.minio.client.MinioRecorder.lambda$minioClientSupplier$0(MinioRecorder.java:13)
	at io.quarkus.arc.runtime.ArcRecorder$4.apply(ArcRecorder.java:129)
	at io.quarkus.arc.runtime.ArcRecorder$4.apply(ArcRecorder.java:126)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.createSynthetic(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.doCreate(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.create(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.get(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.get(Unknown Source)
	at io.cryostat.recordings.Recordings_Bean.doCreate(Unknown Source)
	at io.cryostat.recordings.Recordings_Bean.create(Unknown Source)
	at io.cryostat.recordings.Recordings_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
	at io.cryostat.recordings.Recordings_Observer_onStart_d7d5577ed7c7cf36fa3bfe9393292221b1355418.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:346)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:328)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:82)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:155)
	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:106)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
	... 13 more
Caused by: java.lang.RuntimeException: Unsupported OkHttp library found. Must use okhttp >= 4.8.1
	at io.minio.S3Base.<clinit>(S3Base.java:106)
	... 51 more
Caused by: java.lang.NoSuchMethodError: 'okhttp3.RequestBody okhttp3.RequestBody.create(byte[], okhttp3.MediaType)'
	at io.minio.S3Base.<clinit>(S3Base.java:104)
	... 51 more


2023-04-28 17:22:28,391 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: io.quarkus.dev.appstate.ApplicationStartException: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.dev.appstate.ApplicationStateNotification.waitForApplicationStart(ApplicationStateNotification.java:58)
	at io.quarkus.runner.bootstrap.StartupActionImpl.runMainClass(StartupActionImpl.java:123)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:137)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:447)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:59)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:149)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:104)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:104)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ExceptionInInitializerError
	at io.minio.MinioClient$Builder.<init>(MinioClient.java:2447)
	at io.minio.MinioClient.builder(MinioClient.java:2439)
	at io.quarkiverse.minio.client.MinioClients.createMinioClient(MinioClients.java:70)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at io.quarkiverse.minio.client.MinioClients.getMinioClient(MinioClients.java:53)
	at io.quarkiverse.minio.client.MinioClients.fromName(MinioClients.java:49)
	at io.quarkiverse.minio.client.MinioRecorder.lambda$minioClientSupplier$0(MinioRecorder.java:13)
	at io.quarkus.arc.runtime.ArcRecorder$4.apply(ArcRecorder.java:129)
	at io.quarkus.arc.runtime.ArcRecorder$4.apply(ArcRecorder.java:126)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.createSynthetic(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.doCreate(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.create(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.get(Unknown Source)
	at io.minio.MinioClient_424c51f0b04aa3d15a8d71eb15a366bb7147c621_Synthetic_Bean.get(Unknown Source)
	at io.cryostat.recordings.Recordings_Bean.doCreate(Unknown Source)
	at io.cryostat.recordings.Recordings_Bean.create(Unknown Source)
	at io.cryostat.recordings.Recordings_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
	at io.cryostat.recordings.Recordings_Observer_onStart_d7d5577ed7c7cf36fa3bfe9393292221b1355418.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:346)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:328)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:82)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:155)
	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:106)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
	... 13 more
Caused by: java.lang.RuntimeException: Unsupported OkHttp library found. Must use okhttp >= 4.8.1
	at io.minio.S3Base.<clinit>(S3Base.java:106)
	... 51 more
Caused by: java.lang.NoSuchMethodError: 'okhttp3.RequestBody okhttp3.RequestBody.create(byte[], okhttp3.MediaType)'
	at io.minio.S3Base.<clinit>(S3Base.java:104)
	... 51 more


2023-04-28 17:22:28,394 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to recover after failed start: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:294)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup.handleFailedInitialStart(VertxHttpHotReplacementSetup.java:61)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.startupFailed(RuntimeUpdatesProcessor.java:1150)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:175)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:447)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:59)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:149)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:104)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:78)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:57)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:264)
	... 9 more
Caused by: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleMap(ConfigInstantiator.java:178)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:121)
	... 12 more
Caused by: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3585)
	at java.base/java.lang.Class.getConstructor(Class.java:2271)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:106)
	... 14 more


2023-04-28 17:22:28,394 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to recover after failed start: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:294)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup.handleFailedInitialStart(VertxHttpHotReplacementSetup.java:61)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.startupFailed(RuntimeUpdatesProcessor.java:1150)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:175)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:447)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:59)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:149)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:104)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:78)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:57)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:264)
	... 9 more
Caused by: java.lang.RuntimeException: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:139)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleMap(ConfigInstantiator.java:178)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:121)
	... 12 more
Caused by: java.lang.NoSuchMethodException: java.util.OptionalInt.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3585)
	at java.base/java.lang.Class.getConstructor(Class.java:2271)
	at io.quarkus.runtime.configuration.ConfigInstantiator.handleObject(ConfigInstantiator.java:106)
	... 14 more



--
Press [space] to restart, [e] to edit command line args (currently ''), [:] for the terminal, [h] for more options>
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  29.777 s
[INFO] Finished at: 2023-04-28T17:22:29-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.0.1.Final:dev (default-cli) on project cryostat3: Failed to run: Dev mode process did not complete successfully -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

And checking the project dependency tree, this seems to be indeed a version mismatch:

```
$ mvn dependency:tree | grep -C8 -i okhttp
[INFO] |  \- org.postgresql:postgresql:jar:42.6.0:compile
[INFO] +- io.quarkiverse.minio:quarkus-minio:jar:3.0.0:compile
[INFO] |  \- io.quarkiverse.minio:quarkus-minio-native:jar:3.0.0:compile
[INFO] |     \- io.minio:minio:jar:8.5.2:compile
[INFO] |        +- com.carrotsearch.thirdparty:simple-xml-safe:jar:2.7.1:compile
[INFO] |        +- com.google.guava:guava:jar:31.1-jre:compile
[INFO] |        |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |        |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] |        +- com.squareup.okhttp3:okhttp:jar:3.14.9:compile # <-- here's the problem?
[INFO] |        |  \- com.squareup.okio:okio:jar:1.17.2:compile
[INFO] |        +- org.bouncycastle:bcprov-jdk15on:jar:1.69:compile
[INFO] |        +- org.apache.commons:commons-compress:jar:1.23.0:compile
[INFO] |        \- org.xerial.snappy:snappy-java:jar:1.1.8.4:compile
[INFO] +- org.apache.commons:commons-lang3:jar:3.12.0:compile
[INFO] +- commons-codec:commons-codec:jar:1.15:compile
[INFO] +- commons-io:commons-io:jar:2.11.0:compile
[INFO] +- org.apache.httpcomponents:httpclient:jar:4.5.14:compile
```

--- a/.github/project.yml
+++ b/.github/project.yml
@@ -1,4 +1,4 @@
 name: Quarkus Minio
 release:
-  current-version: 3.0.0
-  next-version: 3.0.1-SNAPSHOT
+  current-version: 3.0.2
+  next-version: 3.0.2-SNAPSHOT

--- a/minio-client/runtime/src/main/java/io/quarkiverse/minio/client/MinioClients.java
+++ b/minio-client/runtime/src/main/java/io/quarkiverse/minio/client/MinioClients.java
@@ -34,13 +34,13 @@ public class MinioClients {
     }
 
     /**
-     * Meant to be used from recorders that create synthetic beans that need access to {@code Datasource}.
-     * In such using {@code Arc.container.instance(DataSource.class)} is not possible because
-     * {@code Datasource} is itself a synthetic bean.
+     * Meant to be used from recorders that create synthetic beans that need access to {@code MinioClient}.
+     * In such using {@code Arc.container.instance(MinioClient.class)} is not possible because
+     * {@code MinioClient} is itself a synthetic bean.
      * <p>
-     * This method relies on the fact that {@code DataSources} should - given the same input -
-     * always return the same {@code AgroalDataSource} no matter how many times it is invoked
-     * (which makes sense because {@code DataSource} is a {@code Singleton} bean).
+     * This method relies on the fact that {@code MinioClients} should - given the same input -
+     * always return the same {@code MinioClient} no matter how many times it is invoked
+     * (which makes sense because {@code MinioClient} is a {@code Singleton} bean).
      * <p>
      * This method is thread-safe
      */

--- a/minio-native/deployment/src/main/java/io/quarkiverse/minio/client/deployment/MinioNativeProcessor.java
+++ b/minio-native/deployment/src/main/java/io/quarkiverse/minio/client/deployment/MinioNativeProcessor.java
@@ -16,7 +16,6 @@ import org.jboss.jandex.ClassInfo;
 import org.jboss.jandex.DotName;
 
 import io.minio.BaseArgs;
-import io.minio.UploadSnowballObjectsArgs;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
@@ -50,7 +49,8 @@ class MinioNativeProcessor {
                 .map(ClassInfo::name)
                 .map(DotName::toString)
                 .collect(Collectors.toList()));
-        reflectionClasses.produce(new ReflectiveClassBuildItem(true, true, classes.toArray(new String[0])));
+        reflectionClasses
+                .produce(ReflectiveClassBuildItem.builder(classes.toArray(new String[0])).fields(true).methods(true).build());
     }
 
     private List<String> getClasses(String packageName, String regexp) {
@@ -80,10 +80,5 @@ class MinioNativeProcessor {
             }
         }
         return null;
-    }
-
-    @BuildStep
-    RuntimeInitializedClassBuildItem randomConfiguration() {
-        return new RuntimeInitializedClassBuildItem(UploadSnowballObjectsArgs.class.getCanonicalName());
     }
 }

--- a/pom.xml
+++ b/pom.xml
@@ -21,11 +21,7 @@
         <failsafe-plugin.version>${version.surefire.plugin}</failsafe-plugin.version>
 
         <quarkus.version>3.0.2.Final</quarkus.version>
-        <minio.version>8.5.2</minio.version>
-        <okhttp.version>4.10.0</okhttp.version>
-        <!-- According to the bom we need matching version of okhttp and okio -->
-        <!-- https://github.com/quarkusio/quarkus/blob/main/bom/application/pom.xml -->
-        <okio.version>3.2.0</okio.version>
+        <minio.version>8.2.2</minio.version>
         <assertj.version>3.24.2</assertj.version>
     </properties>
 
@@ -64,19 +60,6 @@
                 <artifactId>assertj-core</artifactId>
                 <version>${assertj.version}</version>
             </dependency>
-            <!-- According to issue #12680, okhttp was hold back until quarkus-kubernetes-client is updated to 6.x. -->
-            <!-- That was performed in commit f1a50bfe683ea075f67e29321d84508433ed7c63, so since 2.13.0.CR1 -->
-            <dependency>
-                <groupId>com.squareup.okhttp3</groupId>
-                <artifactId>okhttp</artifactId>
-                <version>${okhttp.version}</version>
-            </dependency>
-            <!-- Do we actually need this? AFAIK only okhttp on android dependency on okio! -->
-            <!--dependency>
-                <groupId>com.squareup.okio</groupId>
-                <artifactId>okio</artifactId>
-                <version>${okio.version}</version>
-            </dependency-->
         </dependencies>
     </dependencyManagement>
     <build>

